reference holonix

thedavidmeister opened the pull request:
## PR summary

## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [ ] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

thedavidmeister commented on default.nix:
```diff
@@ -1,18 +1,19 @@
-  openssl = import ./holonix/openssl/config.nix;
-  rust = import ./holonix/rust/config.nix;
-
+ holonix = import (fetchTarball {
```
this is where the magic happens

thedavidmeister commented on default.nix:
```diff
@@ -1,18 +1,19 @@
 let
+ holonix-release-tag = "expecting-bad-ozone";
```
not using semver

Connoropolous commented:
@thedavidmeister can we get a little explainer of what the outcomes of this PR are?

thedavidmeister commented:
@Connoropolous 

- moved the holonix directory to the holonix repo
- reworked the holonix output slightly to be a set rather than a derivation
  - means that downstream consumers can provide a context relevant derivation and use holonix as a variable
  - means that downstream consumers can reference the holonix repo by URL as a tarball
  - core is the first downstream so `foo` becomes `holonix.foo`
  - node js will be the second downstream so we can pull it out
- plan for future is that holonix repo will be refactored to be only the "commons" and downstreams will implement their own workflows e.g. `hc-*`

thedavidmeister commented:
@Connoropolous for example... the conflicts in dist right now should sit in core!

Connoropolous commented:
thanks @thedavidmeister
